packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -173,6 +175,24 @@ export class MultichainBalancesController extends BaseController<
       (balanceUpdate: AccountBalancesUpdatedEventPayload) =>
         this.#handleOnAccountBalancesUpdated(balanceUpdate),
     );
+    // TODO: Maybe add a MultichainAssetsController:accountAssetListUpdated event instead of using the entire state.
+    // Since MultichainAssetsController already listens for the AccountsController:accountAdded, we can rely in it for that event
+    // and not listen for it also here, in this controller, since it would be redundant
+    this.messagingSystem.subscribe(
+      'MultichainAssetsController:stateChange',
+      async (assetsState: MultichainAssetsControllerState) => {
+        for (const accountId in assetsState.accountsAssets) {
+          if (
+            Object.prototype.hasOwnProperty.call(
+              assetsState.accountsAssets,
+              accountId,
+            )
+          ) {
+            await this.updateBalance(accountId);
+          }
```

Comment:
This `if` is not needed, we already are iterating over `assetsState.accountsAssets` no?

```suggestion
          await this.updateBalance(accountId);
```

Reply:
Hmmmm what if a user removes an account? this would trigger a state update; and also trigger this to be executed.
I think that `if` is a safe check to make sure the account is there right?

Reply:
no but that's an "anti-pattern" (mostly required because the foreach loop was wrong 🙈 ), cause if you are iterating over `something`, you don't need to check that each elements are "defined" in `something`

Reply:
To write this differently:

```ts
const map = {
  foo: true,
  bar: false,
  zzz: true,
}

for (const key of Object.keys(map)) {
  // No need to check that `key` is inside `map`
  ...
}
```